src/main/java/net/minecraftforge/client/model/generators/ItemModelBuilder.java

```diff
@@ -23,10 +27,18 @@
  */
 public class ItemModelBuilder extends ModelBuilder<ItemModelBuilder> {
 
+    private static final List<ResourceKey<TrimMaterial>> VANILLA_TRIM_MATERIALS = List.of(TrimMaterials.QUARTZ, TrimMaterials.IRON, TrimMaterials.NETHERITE, TrimMaterials.REDSTONE, TrimMaterials.COPPER, TrimMaterials.GOLD, TrimMaterials.EMERALD, TrimMaterials.DIAMOND, TrimMaterials.LAPIS, TrimMaterials.AMETHYST);
     protected List<OverrideBuilder> overrides = new ArrayList<>();
 
     public ItemModelBuilder(ResourceLocation outputLocation, ExistingFileHelper existingFileHelper) {
         super(outputLocation, existingFileHelper);
+        //add vanilla armor trims to the existing file helper
+        for (ResourceKey<TrimMaterial> trim : VANILLA_TRIM_MATERIALS) {
+            this.existingFileHelper.trackGenerated(new ResourceLocation("boots_trim_" + trim.location().getPath()), PackType.CLIENT_RESOURCES, ".png", "textures/trims/item");
```

Comment:
From my understanding, the prefix is intended to be the general root folder of a given "resource type". In this case that would mean that the prefix should just be `textures` and the rest of your current prefix should be included in the file name, at which point it would be cleaner to use the existing `ResourceType` specified in `ModelProvider.TEXTURE` instead of specifying the equivalent pack type, prefix and suffix manually. Making said `ResourceType` field public to allow modders to use it outside of a `ModelProvider` (i.e. in a `SpriteSourceProvider` or a `BlockStateProvider`) would be a good fit for this PR as well in my opinion.

Reply:
I didnt technically need to make it public since ModelProvider is in the same package but I did anyway since you mentioned that last part. Will revert that back if someone isnt happy about it